<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1366</width>
    <height>768</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">/* Groupbox */

QGroupBox {
    border: 1px solid #DDD;
    border-radius: 9px;
    margin-top: 9px;
}

QGroupBox::title {
    subcontrol-origin: margin;
    left: 10px;
    padding: 0 3px 0 3px;
}</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="leftMargin">
     <number>6</number>
    </property>
    <property name="topMargin">
     <number>6</number>
    </property>
    <property name="rightMargin">
     <number>6</number>
    </property>
    <property name="bottomMargin">
     <number>6</number>
    </property>
    <item>
     <widget class="QGroupBox" name="OptionsGB">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>220</width>
        <height>0</height>
       </size>
      </property>
      <property name="title">
       <string>Options</string>
      </property>
      <widget class="QPushButton" name="LoadOBJ">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>30</y>
         <width>181</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Load OBJ file</string>
       </property>
      </widget>
      <widget class="QSlider" name="set_ref_line_size">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>710</y>
         <width>171</width>
         <height>22</height>
        </rect>
       </property>
       <property name="mouseTracking">
        <bool>true</bool>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>45</number>
       </property>
       <property name="tracking">
        <bool>true</bool>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
      <widget class="QLabel" name="RefLineLabel">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>690</y>
         <width>171</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Isophote line thickness</string>
       </property>
      </widget>
      <widget class="QCheckBox" name="toggle_wf_mode">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>630</y>
         <width>181</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Wireframe mode</string>
       </property>
      </widget>
      <widget class="QCheckBox" name="toggle_ref_lines">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>660</y>
         <width>181</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Show isophotes</string>
       </property>
      </widget>
      <widget class="QLabel" name="SubdivLabel">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>500</y>
         <width>181</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Loop subdivision steps</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="SubdivSteps">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>520</y>
         <width>181</width>
         <height>22</height>
        </rect>
       </property>
       <property name="maximum">
        <number>8</number>
       </property>
      </widget>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>550</y>
         <width>181</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>View-mode (use arrows)</string>
       </property>
      </widget>
      <widget class="QComboBox" name="viewModeBox">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>570</y>
         <width>181</width>
         <height>22</height>
        </rect>
       </property>
       <item>
        <property name="text">
         <string>Loop subdivision</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>QAS</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Loop and QAS side-by-side</string>
        </property>
       </item>
      </widget>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>70</y>
         <width>181</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Drag mouse to change camera view.</string>
       </property>
       <property name="wordWrap">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QSlider" name="tessellationLevelSlider">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>370</y>
         <width>181</width>
         <height>16</height>
        </rect>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>64</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>320</y>
         <width>181</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>QAS tessellation level</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_4">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>240</y>
         <width>181</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>QAS subdivision offset</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="qasSubdivOffsetBox">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>260</y>
         <width>181</width>
         <height>23</height>
        </rect>
       </property>
      </widget>
      <widget class="QCheckBox" name="adaptiveTessellationBox">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>400</y>
         <width>181</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Adaptive tessellation</string>
       </property>
      </widget>
      <widget class="QComboBox" name="backgroundBox">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>140</y>
         <width>181</width>
         <height>22</height>
        </rect>
       </property>
       <item>
        <property name="text">
         <string>Transparent</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Black</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Dark gray</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>White</string>
        </property>
       </item>
      </widget>
      <widget class="QLabel" name="label_5">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>120</y>
         <width>181</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Background color:</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_6">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>180</y>
         <width>101</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>QAS polygons:</string>
       </property>
      </widget>
      <widget class="QLabel" name="qasPolygonCountLabel">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>180</y>
         <width>91</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>0</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="label_7">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>200</y>
         <width>101</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Loop polygons:</string>
       </property>
      </widget>
      <widget class="QLabel" name="loopPolygonCountLabel">
       <property name="geometry">
        <rect>
         <x>119</x>
         <y>200</y>
         <width>91</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>0</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
      <widget class="QCheckBox" name="normalCurvatureBox">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>290</y>
         <width>191</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Show normal-curvature</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_8">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>430</y>
         <width>71</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Zoom</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_9">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>450</y>
         <width>71</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Curvature</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_10">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>470</y>
         <width>60</width>
         <height>14</height>
        </rect>
       </property>
       <property name="text">
        <string>Normal</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="zoomTessSlider">
       <property name="geometry">
        <rect>
         <x>140</x>
         <y>430</y>
         <width>61</width>
         <height>23</height>
        </rect>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="curvTessSlider">
       <property name="geometry">
        <rect>
         <x>140</x>
         <y>450</y>
         <width>61</width>
         <height>23</height>
        </rect>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="normTessSlider">
       <property name="geometry">
        <rect>
         <x>140</x>
         <y>470</y>
         <width>61</width>
         <height>23</height>
        </rect>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
      </widget>
      <widget class="QLabel" name="label_11">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>340</y>
         <width>31</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Min:</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="qasTessMinSlider">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>340</y>
         <width>48</width>
         <height>23</height>
        </rect>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>64</number>
       </property>
      </widget>
      <widget class="QLabel" name="label_12">
       <property name="geometry">
        <rect>
         <x>119</x>
         <y>340</y>
         <width>31</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Max:</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="qasTessMaxSlider">
       <property name="geometry">
        <rect>
         <x>150</x>
         <y>340</y>
         <width>48</width>
         <height>23</height>
        </rect>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>64</number>
       </property>
       <property name="value">
        <number>64</number>
       </property>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="MainView" name="MainDisplay"/>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MainView</class>
   <extends>QOpenGLWidget</extends>
   <header>mainview.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
